capstone/src/main/java/com/google/sps/servlets/RegisterDriver.java

```diff
@@ -27,39 +30,44 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         List<Driver> drivers = new ArrayList<>();
-        Query query = new Query("Driver").addSort("timestamp", SortDirection.DESCENDING);
+        Query queryD = new Query("Driver").addSort("timestamp", SortDirection.DESCENDING);
 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(query);
+        PreparedQuery resultsD = datastore.prepare(queryD);
 
-        for (Entity entity : results.asIterable()) {  
+        for (Entity entity : resultsD.asIterable()) {  
             String first = (String)entity.getProperty("first");
             String last = (String)entity.getProperty("last");
             String day = (String)entity.getProperty("day");
             String times = (String)entity.getProperty("times");
             long seats = (long)entity.getProperty("seats");
             long timestamp = (long)entity.getProperty("timestamp");
-            long id = (long)entity.getProperty("id");
+            Long id = (Long)entity.getProperty("id");
+            if (id == null) {
+               id = 0L; 
+            }
             Driver driver = new Driver(first,last,day,times,seats,timestamp,id);
             drivers.add(driver);
         }
         
-        Gson gson = new Gson();
+        String gson = new Gson().toJson(drivers); 
         response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(drivers));
+        String both = "{'drivers':" + gson + "}"; 
+        JSONObject json = new JSONObject(both);
+        response.getWriter().println(json.toString());
     }
+
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String first = getParameter(request, "first", "");
         String last = getParameter(request, "last", "");
         String day = getParameter(request, "day", "");
         String times = getParameter(request, "times", "");
         long seats = Long.valueOf(getParameter(request, "seats", ""));
         long timestamp = System.currentTimeMillis();
-        long id = (long)(Math.random() * (Long.MAX_VALUE - 0 + 1) + 0);
+        long id = (long)((Math.random() * (Long.MAX_VALUE - 0 + 1) + 0) * -1);
```

Comment:
What's going on here with the - 0 + 1 * -1?

Reply:
The -1 was to make sure the id's were positive!

Reply:
I think it's only negative because of the Long.MAX_VALUE + 1 part, which is the same as Long.MIN_VALUE since it wraps around, so we're generating a random number between -2**63 and 0, then multiplying by -1 to make it positive. If you take out the extra steps it's just:

"Math.random() * Long.MAX_VALUE"